R/support.R

```diff
@@ -2,13 +2,14 @@
 #'
 #' @param x A list of prototype vectors defining the distribution type.
 #' @param limits A list of value limits for the distribution.
+#' @param closed A list of logical(2L) indicating whether the limits are closed.
 #'
-new_support_region <- function(x, limits = NULL) {
-  vctrs::new_rcrd(list(x = x, lim = limits), class = "support_region")
+new_support_region <- function(x, limits = NULL, closed = list(c(TRUE, TRUE))) {
```

Comment:
```suggestion
new_support_region <- function(x = numeric(), limits = list(), closed = list()) {
```
The convention for vctrs creation functions is to produce a length 0 output by default (which admittedly I haven't done here). That way `new_support_region()` behaves similarly to `numeric()` and `list()` by giving a length 0 output for the right type.